hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsOutputStream.java

```diff
@@ -353,14 +353,16 @@ public synchronized void write(final byte[] data, final int off, final int lengt
       throw new PathIOException(path, ERR_WRITE_WITHOUT_LEASE);
     }
 
+    if (length == 0) {
+      return;
+    }
+
     AbfsBlock block = createBlockIfNeeded(position);
     // Put entry in map with status as NEW which is changed to SUCCESS when successfully appended.
-    mapLock.lock();
     try {
-      if (length > 0) {
-        map.put(block.getBlockId(), BlockStatus.NEW);
-        orderedBlockList.add(block.getBlockId());
-      }
+      mapLock.lock();
```

Comment:
We want to release the lock in a finally block always as that is a good practice, hence had a try catch here.

Reply:
length > 0 check is because we don't want a network call to go when we are issuing writes with no length.

Reply:
> We want to release the lock in a finally block always as that is a good practice, hence had a try catch here.

the block inside try will not throw any exceptions as its just put / add. If there is not possiblity of exception, we dont need try-catch-finally.

Reply:
> length > 0 check is because we don't want a network call to go when we are issuing writes with no length.

API call fire doesnt depend on this. Inside `uploadBlockAsync`, there is a check on `!blockToUpload.hasData()`. If the `hasData()` returns false, we dont fire the call, neither we submit the task. Maybe there would be need to let length==0 flow in (as per trunk code)? Lets check the feasiblity after this PR. Lets not stop pr for this.

Reply:
Will still want to keep the code inside try and finally :- Releasing locks inside a finally block is considered good practice in Java, especially when dealing with concurrent programming and synchronization. The finally block ensures that the code within it gets executed regardless of whether an exception is thrown or not. This is important for maintaining the correctness and safety of your program when dealing with locks and critical sections.

Reply:
The flush call would still go, we can prevent a flush call by returning from here only